File: Linked_List/160. Intersection of Two Linked Lists.py
class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
        
        if headA is None or headB is None:
            return None
        
        itr1,itr2=headA,headB
        
        while itr1!=itr2:
            if itr1 is None:
                itr1=headB
            else:
                itr1=itr1.next
            if itr2 is None:
                itr2=headA
            else:
                itr2=itr2.next
                
        return itr1

File: Linked_List/Java/160. Intersection of Two Linked Lists.java
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode itr1=headA;
        ListNode itr2=headB;

        while(itr1!=itr2)
        {
            if(itr1!=null) 
            {
                itr1=itr1.next;
            }
            else
            {
                itr1=headB;
            }

             if(itr2!=null)
             {
                 itr2=itr2.next;
             }
             else
             {
                 itr2=headA;
             }
        }
        return itr1;
    }
}